admin/Default/1.6/universe_create_save_processing.php

```diff
@@ -167,39 +167,27 @@
 	$galaxies =& SmrGalaxy::getGameGalaxies($var['game_id']);
 	$galaxy =& SmrGalaxy::getGalaxy($var['game_id'],$var['gal_on']);
 	$galSectors =& $galaxy->getSectors();
-	//get totals
-	$numberOfPlanets=0;
 	foreach ($galSectors as &$galSector) {
 		if($galSector->hasPlanet()) {
 			$galSector->removePlanet();
 		}
 	}
-	$numberOfPlanets = $_POST['Uninhab'];
 //	$numberOfNpcPlanets = $_POST['NPC'];
-	$updatePlan = array();
-	$updatePlan['Uninhab'] = $num_uninhab;
-	$updatePlan['NPC'] = $num_npc;
-	
-	for ($i=1;$i<=$numberOfPlanets;$i++) {
-		$galSector =& $galSectors[array_rand($galSectors)];
-		while ($galSector->hasPlanet()) $galSector =& $galSectors[array_rand($galSectors)]; //1 per sector
 
-		$galSector->createPlanet();
-		//inhabitable between 7 and 10 days (from start of game)
-//		$updatePlan[$sector]['Inhabitable'] = $start_time + mt_rand(7,10) * 3600 * 24;
-//		$updatePlan[$sector]['Owner'] = 0;
-//		$updatePlan[$sector]['Owner Type'] = 'Player';
+	$allowedTypeIDs = array();
+	$db->query('SELECT planet_type_id FROM planet_type');
+	while ($db->nextRecord()) {
+		$allowedTypeIDs[] = $db->getInt('planet_type_id');
+	}
+
+	foreach ($allowedTypeIDs as $planetTypeID) {
+		$numberOfPlanets = $_POST['type' . $planetTypeID];
+		for ($i=1;$i<=$numberOfPlanets;$i++) {
```

Comment:
Could you change this to
```
for ($i = 0; $i < $numberOfPlanets; $i++) {
```
please.  As `$i` isn't used it's weird to see it start from 1

Reply:
This line was there before I made any changes (line 183), and `$i` wasn't used in the previous version either. I can change it if you want, but the rest of this file does the exact same thing (which would make this line the odd one out). But if it were my choice, I'd rather leave it as is (to address in a later cleanup of this file) to avoid entangling functionality changes with code cleanup. Let me know!

Reply:
Yeah, I realised it wasn't a change by you - I didn't realise it's used throughout the rest of this file though, we can't leave it as is so it's at least consistent within the file